### DISCUSSION TITLE
How to use msw in the indie-stack with the v2_dev flag

### DISCUSSION THREAD

**Discussion by MattyBalaam** [remix-run/remix#6647] [Q&A] [answered]
At the moment if you run the indie-stack with the `v2_dev` flag on remix nightly (or `unstable_dev`) for older versions then no mocking happens, it just silently fails.

<img width="1461" alt="Screenshot 2023-06-20 at 14 30 58" src="https://github.com/remix-run/remix/assets/1246923/546fe769-e705-4020-b354-1597067567ac">

Expected:
<img width="612" alt="Screenshot 2023-06-20 at 14 29 40" src="https://github.com/remix-run/remix/assets/1246923/a5493ffd-3624-438a-bb3d-6a18666c6fbc">

Actual:
<img width="864" alt="Screenshot 2023-06-20 at 14 30 06" src="https://github.com/remix-run/remix/assets/1246923/4fdfaa7a-4216-4a07-8977-aba27956ed7a">

**Comment by MattyBalaam** [comment, marked as answer]
I had a chat with @pcattori who was very helpful: https://discord.com/channels/770287896669978684/1121004019008147527 and provide a couple of solutions.

Either change the script structure:
```
{
  "scripts": {
    "dev": "remix dev -c 'npm run dev:app'",
    "dev:app": "binode --require ./mocks -- @remix-run/serve:remix-serve ./build",
  }
}
```

or move the server start into `entry.server.tsx`

There is a fix for a small issue: https://github.com/remix-run/remix/pull/6663

Note that `./build` needs to be the folder you are building `serverBuildPath`

**Reply by pcattori** [reply]
Instructions will be included in docs for next release (https://github.com/remix-run/remix/pull/6669)
